src/views/List.jsx

```diff
@@ -1,13 +1,50 @@
+import { useState } from 'react';
+
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchTerm, setSearchTerm] = useState('');
+
+	const filterList = (list) => {
+		return list.filter(({ name }) => {
+			return name
+				.toLowerCase()
+				.trim()
+				.replace(/\s+/g, ' ')
+				.includes(searchTerm.toLowerCase().replace(/\s+/g, ' ').trim());
```

Comment:
I noticed that `searchTerm` has the same methods applied to it as `name`. It might be good to create a helper function to reduce the duplication a little bit, like:

```javascript
const clean = str => str.toLowerCase().trim().replace(/\s+/g, ' ');

...
clean(name).includes(clean(searchTerm))
```

Reply:
Good point, thanks! It's now pulled out to sit inside filterList to limit scope.